yb-voyager/cmd/exportData_test.go

```diff
@@ -0,0 +1,660 @@
+	}
+
+	expectedMissingTables := []sqlname.NameTuple{
+		getNameTuple("public.datatypes1"),
+		getNameTuple("public.foreign_test"),
+	}
+
+	expectedExtraTables := []sqlname.NameTuple{
+		getNameTuple("p1.london"),
+		getNameTuple("p1.sydney"),
+	}
+
+	assertGuardrailsChecksForMissingAndExtraTablesInSubsequentRun(t, expectedMissingTables, expectedExtraTables, firstRunTableList, rootTables)
+	getInitialTableistAndAssertExpectedResult(t, firstRunTableList, firstRunPartitionsToRootMap)
+
+	//case4: getInitialTableList for subsequent run with  start-clean false and basic with only exclude-table-list flag
+	startClean = false
+	//--exclude-table-list sales_region,foreign_test
+	source.TableList = ""
+	source.ExcludeTableList = "sales_region,foreign_test"
+	utils.DoNotPrompt = true
+
+	expectedMissingTables2 := []sqlname.NameTuple{
+		getNameTuple("p1.boston"),
+		getNameTuple("public.foreign_test"),
+	}
+
+	expectedExtraTables2 := []sqlname.NameTuple{}
+
+	assertGuardrailsChecksForMissingAndExtraTablesInSubsequentRun(t, expectedMissingTables2, expectedExtraTables2, firstRunTableList, rootTables)
+	getInitialTableistAndAssertExpectedResult(t, firstRunTableList, firstRunPartitionsToRootMap)
+
+	//case5: getInitialTableList for subsequent run with  start-clean false and basic with table-list and exclude-table-list flags
+	startClean = false
+	//--table-list test_partitions_sequences,foreign_test,p1.london --exclude-table-list sales_region
+	source.TableList = "test_partitions_sequences,foreign_test,p1.london"
+	source.ExcludeTableList = "sales_region"
+	utils.DoNotPrompt = true
+
+	expectedMissingTables3 := []sqlname.NameTuple{
+		getNameTuple("p1.boston"),
+		getNameTuple("public.datatypes1"),
+	}
+
+	expectedExtraTables3 := []sqlname.NameTuple{}
+
+	assertGuardrailsChecksForMissingAndExtraTablesInSubsequentRun(t, expectedMissingTables3, expectedExtraTables3, firstRunTableList, rootTables)
+	getInitialTableistAndAssertExpectedResult(t, firstRunTableList, firstRunPartitionsToRootMap)
+
+	//case6: getInitialTableList for subsequent run with  start-clean false and basic with different set of table-list and exclude-table-list flags
+	startClean = false
+	//--table-list test_partitions_sequences,p1.boston,foreign_test,p1.london,datatypes1 --exclude-table-list datatypes1,test_partitions_sequences_s
+	source.TableList = "test_partitions_sequences,p1.boston,foreign_test,p1.london,datatypes1"
+	source.ExcludeTableList = "datatypes1,test_partitions_sequences_s"
+	utils.DoNotPrompt = true
+
+	expectedMissingTables4 := []sqlname.NameTuple{
+		getNameTuple("public.test_partitions_sequences_s"),
+		getNameTuple("public.datatypes1"),
+	}
+
+	expectedExtraTables4 := []sqlname.NameTuple{
+		getNameTuple("p1.london"),
+	}
+
+	assertGuardrailsChecksForMissingAndExtraTablesInSubsequentRun(t, expectedMissingTables4, expectedExtraTables4, firstRunTableList, rootTables)
+	getInitialTableistAndAssertExpectedResult(t, firstRunTableList, firstRunPartitionsToRootMap)
+}
```

Comment:
test idea: ensure that 
first run: `foo*` (foo1, foo2)
second run : `foo1,foo2` does not result in any guardrails thrown.

Reply:
yeah, I had it in mind for adding more tests around glob pattern support in the table list flags, will add it in a separate PR.